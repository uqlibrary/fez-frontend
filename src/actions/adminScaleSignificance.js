import * as actions from './actionTypes';

export function updateAdminScaleSignificance(items) {
    return dispatch => {
        dispatch({ type: actions.ADMIN_SCALE_SIGNIFICANCE_UPDATED, payload: items });
    };
}
