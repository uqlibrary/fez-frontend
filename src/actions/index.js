export * from './academic';
export * from './account';
export * from './app';
export * from './authors';
export * from './batchImport';
export * from './bulkUpdates';
export * from './claimPublications';
export * from './controlledVocabularies';
export * from './deleteRecord';
export * from './doi';
export * from './exportPublications';
export * from './favouriteSearch';
export * from './fixRecord';
export * from './incompleteRecords';
export * from './issnLinks';
export * from './journals';
export * from './manageAuthors';
export * from './manageUsers';
export * from './myEditorialAppointments';
export * from './orcidSync';
export * from './organisationalUnits';
export * from './publications';
export * from './records';
export * from './search';
export * from './thirdPartyLookupTool';
export * from './viewRecord';
export * from './viewCommunitiesCollections';
export * from './adminScaleSignificance';
export * from './viewControlledVocab';
