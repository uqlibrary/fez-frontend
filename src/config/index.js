export {
    API_URL,
    APP_URL,
    AUTH_URL_LOGIN,
    AUTH_URL_LOGOUT,
    SESSION_COOKIE_NAME,
    SESSION_USER_GROUP_COOKIE_NAME,
    TOKEN_NAME,
    ORCID_BASE_URL,
    ORCID_CLIENT_ID,
    ORCID_AUTHORIZATION_URL,
    GOOGLE_MAPS_API_URL,
    GOOGLE_MAPS_API_CHINA_URL,
    PATH_PREFIX,
    DATASET_ACCESS_CONDITIONS_OPTIONS,
} from './general';

export { api, generateCancelToken, sessionApi } from './axios';
export * as validation from './validation';
export * as general from './general';
export { publicationTypes, contentIndicators, numberToWords } from './general';
export * as routes from './routes';
export * as openAccessConfig from './openAccess';
export { default as incompleteRecord } from './incompleteRecord';
export { viewRecordsConfig } from './viewRecord';
export { trendingPublicationsConfig } from './trendingPublications';
export { claimRecordConfig } from './claimRecord';
export { mui1theme, adminTheme } from './theme';
export * as BULK_UPDATES from './bulkUpdates';
export { pathConfig } from './pathConfig';
export { viewJournalConfig } from './viewJournal';
export { TRANSITION_COHORT } from './thesisSubmissionTransition';

export { default as imageGalleryConfig } from './imageGalleryConfig';
export { communityCollectionsConfig } from './communityCollections';
export { controlledVocabConfig } from './controlledVocabConfig';
