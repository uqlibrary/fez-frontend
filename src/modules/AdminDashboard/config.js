import React from 'react';
import moment from 'moment';

import Chip from '@mui/material/Chip';
import Typography from '@mui/material/Typography';

import { ExternalLink } from 'modules/SharedComponents/ExternalLink';

export const COLOURS = { assigned: '#338CFA', unassigned: '#B60DCE' };

export const LINK_UNPROCESSED_WORKS =
    'https://espace.library.uq.edu.au/records/search?page=1&pageSize=20&sortBy=score&sortDirection=Desc&searchQueryParams%5Ball%5D%5Bvalue%5D=ismemberof%3A+UQ%3A218198&searchQueryParams%5Ball%5D%5Blabel%5D=&searchMode=advanced';

export const animationTemplate = (i, duration, delay) =>
    `animateFadeIn ${duration}ms ease-out ${delay * (i + 1)}ms forwards`;

export const MENUACTIONS = {
    EDIT: 'EDIT',
    DELETE: 'DELETE',
    UP: 'MOVEUP',
    TOP: 'MOVETOP',
    DOWN: 'MOVEDOWN',
    BOTTOM: 'MOVEBOTTOM',
};

export const VIEWMODES = {
    VIEW: 'VIEW',
    ADD: 'ADD',
    EDIT: 'EDIT',
    DELETE: 'DELETE',
};
export const VIEWADMINPANELMODES = [VIEWMODES.ADD, VIEWMODES.EDIT, VIEWMODES.DELETE];
export const REORDERING = [MENUACTIONS.TOP, MENUACTIONS.UP, MENUACTIONS.BOTTOM, MENUACTIONS.DOWN];

export const DEFAULT_DATEPICKER_INPUT_FORMAT = 'DD/MM/YYYY';
export const DEFAULT_DATE_FORMAT = 'Do MMMM YYYY';
export const DEFAULT_DATE_FORMAT_WITH_TIME = 'Do MMMM YYYY hh:mm';
export const DEFAULT_DATE_FORMAT_WITH_TIME_24H = 'Do MMMM YYYY HH:mm';
export const DEFAULT_DATE_FORMAT_WITH_TIME_24H_SECONDS = 'Do MMMM YYYY HH:mm:ss';
export const DEFAULT_SERVER_DATE_FORMAT = 'YYYY-MM-DD HH:mm:ss';

export const SYSTEM_ALERT_ACTION = {
    ASSIGN: 'ASSIGN',
    RESOLVE: 'RESOLVE',
};

export const REPORT_TYPE = {
    systemalertlog: 1,
    workshistory: 2,
    workiddupe: 3,
    scopusiddupe: 4,
    doidupe: 5,
};

export const DEFAULT_SORTING = {
    alerts: [{ field: 'sat_created_date', sort: 'asc' }],
    systemalertlog: [{ field: 'sat_created_date', sort: 'asc' }],
    workshistory: [{ field: 'pre_date', sort: 'asc' }],
};

export const isUrl = str => {
    try {
        const newUrl = new URL(str);
        return newUrl.protocol === 'http:' || newUrl.protocol === 'https:';
    } catch (err) {
        return false;
    }
};

export const defaultLegacyReportOption = { sel_id: 0, sel_title: '', sel_description: '' };

export const optionDoubleRowRender = (props, option) => {
    return (
        <li
            {...props}
            style={{
                flexDirection: 'column',
                justifyContent: 'left',
                alignItems: 'flex-start',
                fontWeight: 400,
            }}
            data-testid={props.id}
        >
            <Typography variant="body1" color="textPrimary">
                {option.sel_title}
            </Typography>
            <Typography variant="body1" color="textSecondary">
                {option.sel_description}
            </Typography>
        </li>
    );
};

export const getReportTypeFromValue = value => Object.entries(REPORT_TYPE).find(arr => arr[1] === value)?.[0];
export const getDefaultSorting = reportType => DEFAULT_SORTING?.[reportType] || [];

export const getFormattedServerDate = (dateStr, format = DEFAULT_DATE_FORMAT) =>
    (dateStr && moment(dateStr, DEFAULT_SERVER_DATE_FORMAT).format(format)) || '';

export const getSystemAlertColumns = (locale, users) => {
    const alertStatus = locale.alertStatus;
    const alertStatusOption = Object.values(alertStatus);
    return [
        {
            field: 'sat_created_date',
            headerName: locale.columns.createdDate,
            width: 200,
            type: 'date',
            valueGetter: value => moment(value, DEFAULT_SERVER_DATE_FORMAT).toDate(),
            valueFormatter: value => moment(value).format(DEFAULT_DATE_FORMAT_WITH_TIME_24H),
        },
        { field: 'sat_title', headerName: locale.columns.topic, flex: 1 },
        {
            field: 'status',
            headerName: locale.columns.status,
            width: 160,
            valueGetter: (_, row) =>
                !!row.sat_assigned_to
                    ? users.find(user => user.id === row.sat_assigned_to)?.preferred_name ?? alertStatus.UNKNOWN
                    : alertStatus.UNASSIGNED,
            renderCell: params => (
                <Chip
                    data-testid={`alert-status-${params.id}`}
                    label={params.value}
                    variant="filled"
                    size="small"
                    color={alertStatusOption.includes(params.value) ? 'default' : 'primary'}
                />
            ),
        },
    ];
};

export const getDisplayReportColumns = ({ locale, actionState, params }) => {
    const report = actionState?.displayReport?.value || getReportTypeFromValue(params.report_type);

    const txt = locale.columns[report];
    switch (report) {
        case 'workshistory':
            return [
                { field: 'pre_id', headerName: txt.id, width: 150, exportOnly: true, exportOrder: 0 },
                { field: 'pre_pid', headerName: txt.pid, width: 150, order: 0, exportOrder: 1 },
                {
                    field: 'pre_date',
                    headerName: txt.dateCreated,
                    width: 150,
                    type: 'date',
                    valueGetter: value => (!!value && moment(value, DEFAULT_SERVER_DATE_FORMAT).toDate()) || null,
                    valueFormatter: value => {
                        if (value === null || value === '') return '';
                        return moment(value).format(DEFAULT_DATE_FORMAT);
                    },
                    order: 1,
                    exportOrder: 5,
                },
                {
                    field: 'rek_date',
                    headerName: txt.pubDate,
                    width: 150,
                    type: 'date',
                    valueGetter: value =>
                        (!!value && moment(value, DEFAULT_SERVER_DATE_FORMAT).format(DEFAULT_DATE_FORMAT)) || '',
                    exportOnly: true,
                    exportOrder: 4,
                },
                { field: 'rek_genre', headerName: txt.genre, minWidth: 300, flex: 1, order: 2, exportOrder: 2 },
                { field: 'rek_subtype', headerName: txt.subtype, minWidth: 300, flex: 1, order: 3, exportOrder: 3 },
                { field: 'usr_username', headerName: txt.user, width: 150, order: 4, exportOrder: 6 },
                { field: 'pre_detail', headerName: txt.action, minWidth: 600, flex: 1, order: 5, exportOrder: 7 },
            ];
        default:
            const systemIdParam = actionState?.systemAlertId || params?.record_id || '';
            const cols = [
                { field: 'sat_id', headerName: txt.id, order: 0, exportOrder: 0 },
                {
                    field: 'sat_created_date',
                    headerName: txt.dateCreated,
                    width: 150,
                    type: 'date',
                    valueGetter: value => (!!value && moment(value, DEFAULT_SERVER_DATE_FORMAT).toDate()) || null,
                    valueFormatter: value => {
                        if (value === null || value === '') return '';
                        return moment(value).format(DEFAULT_DATE_FORMAT);
                    },
                    order: 1,
                    exportOrder: 1,
                },
                {
                    field: 'assigned_to_username',
                    headerName: 'Assigned username',
                    width: 150,
                    exportOrder: 2,
                    exportOnly: true,
                },
                {
                    field: 'assigned_to_full_name',
                    headerName: txt.assignedTo,
                    width: 150,
                    exportOrder: 3,
                    order: 2,
                },
                {
                    field: 'sat_assigned_date',
                    headerName: txt.assignedDate,
                    width: 150,
                    type: 'date',
                    valueGetter: value => (!!value && moment(value, DEFAULT_SERVER_DATE_FORMAT).toDate()) || null,
                    valueFormatter: value => {
                        if (value === null || value === '') return '';
                        return moment(value).format(DEFAULT_DATE_FORMAT);
                    },
                    order: 3,
                    exportOrder: 4,
                },
                {
                    field: 'resolved_by_username',
                    headerName: 'Resolved username',
                    width: 150,
                    exportOrder: 5,
                    exportOnly: true,
                },
                {
                    field: 'resolved_by_full_name',
                    headerName: txt.resolvedBy,
                    width: 150,
                    order: 4,
                    exportOrder: 6,
                },
                {
                    field: 'sat_resolved_date',
                    headerName: txt.resolvedDate,
                    width: 150,
                    type: 'date',
                    valueGetter: value => (!!value && moment(value, DEFAULT_SERVER_DATE_FORMAT).toDate()) || null,
                    valueFormatter: value => {
                        if (value === null || value === '') return '';
                        return moment(value).format(DEFAULT_DATE_FORMAT);
                    },
                    order: 5,
                    exportOrder: 7,
                },
                { field: 'sat_title', headerName: txt.title, minWidth: 400, flex: 1, order: 6, exportOrder: 8 },
                {
                    field: 'sat_link',
                    headerName: txt.link,
                    minWidth: 400,
                    flex: 1,
                    renderCell: row =>
                        isUrl(row.value) ? (
                            <ExternalLink id={`link_${row.id}`} href={row.value} inline>
                                {row.value}
                            </ExternalLink>
                        ) : (
                            row.value
                        ),
                    order: 8,
                    exportOrder: 9,
                },
            ];

            if (systemIdParam !== '') {
                cols.push({
                    field: 'sat_content',
                    headerName: txt.content,
                    minWidth: 1000,
                    flex: 1,
                    order: 7,
                    exportOrder: 10,
                });
            }
            return cols;
    }
};
