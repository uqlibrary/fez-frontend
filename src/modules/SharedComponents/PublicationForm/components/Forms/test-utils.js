import { useForm } from 'react-hook-form';
import { render, WithReduxStore } from 'test-utils';
import React from 'react';
import { formValues } from '../PublicationForm';

export const setupWrapper = (Form, props) => {
    const Wrapper = () => {
        const {
            control,
            formState: { isSubmitting },
        } = useForm();

        // eslint-disable-next-line react/prop-types
        return <Form {...{ control, isSubmitting, ...props, formValues: formValues(props?.formValues || {}) }} />;
    };

    return render(
        <WithReduxStore>
            <Wrapper />
        </WithReduxStore>,
    );
};
