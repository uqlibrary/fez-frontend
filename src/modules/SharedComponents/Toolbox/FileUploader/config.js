export const FILE_ACCESS_CONDITION_CLOSED = 1;
export const FILE_ACCESS_CONDITION_OPEN = 5;
export const FILE_ACCESS_CONDITION_INHERIT = 99;
export const FILE_SECURITY_POLICY_ADMIN = 1;
export const FILE_SECURITY_POLICY_EVIDENCE_ASSESSORS = 3;
export const FILE_SECURITY_POLICY_PUBLIC = 5;

export const FILE_ACCESS_OPTIONS = [
    { text: 'Open Access', value: FILE_ACCESS_CONDITION_OPEN },
    { text: 'Closed Access', value: FILE_ACCESS_CONDITION_CLOSED },
];

export const INHERIT_OPTION = {
    text: 'Inherit from parent collection',
    value: FILE_ACCESS_CONDITION_INHERIT,
};

export const SIZE_BASE = 1000;

export const DEFAULT_FILE_UPLOAD_LIMIT = 10;
export const DEFAULT_MAX_FILE_SIZE = 5;

export const SIZE_UNIT_B = 'B';
export const SIZE_UNIT_K = 'K';
export const SIZE_UNIT_M = 'M';
export const SIZE_UNIT_G = 'G';

export const FILE_META_KEY_ACCESS_CONDITION = 'access_condition_id';
export const FILE_META_KEY_EMBARGO_DATE = 'date';
export const FILE_META_KEY_DESCRIPTION = 'description';
export const FILE_META_KEY_SECURITY_POLICY = 'security_policy';

export const SIZE_UNITS = [SIZE_UNIT_B, SIZE_UNIT_K, SIZE_UNIT_M, SIZE_UNIT_G];

export const FILE_NAME_RESTRICTION = /^(?=^\S*$)(?=^[a-z\d\-_]+(\.\d{3}|\.r\d{2,3}|\.part\d{1,3})?\.[^\.]+$)(?=.{1,45}$)(?!(web_|preview_|thumbnail_|stream_|fezacml_|presmd_|\d))[a-z\d\-_\.]+/;

/**
 * When updating the following list, please make sure to update it as well in the following projects:
 * - https://github.com/uqlibrary/fez-frontend
 * - https://github.com/uqlibrary/lambda-fez-upload-handler-trigger
 * - https://github.com/uqlibrary/api
 * - https://github.com/uqlibrary/fez
 *
 * Note: ext -> mimetype is a 1-n relation, but we only care about one value
 */
export const MIME_TYPE_WHITELIST = {
    pdf: 'application/pdf',
    jpe: 'image/jpeg',
    jpeg: 'image/jpeg',
    jpg: 'image/jpeg',
    gif: 'image/gif',
    tif: 'image/tiff',
    tiff: 'image/tiff',
    png: 'image/png',
    zip: 'application/zip',
    tar: 'application/x-tar',
    gz: 'application/gzip',
    '7z': 'application/x-7z-compressed',
    rar: 'application/x-rar-compressed',
    mk3d: 'video/x-matroska',
    mks: 'video/x-matroska',
    mkv: 'video/x-matroska',
    qt: 'video/quicktime',
    mov: 'video/quicktime',
    mp4: 'video/mp4',
    mp4v: 'video/mp4',
    mp3: 'audio/mp3',
    mxf: 'application/mxf',
    wav: 'audio/wav',
    mpe: 'video/mpeg',
    mpeg: 'video/mpeg',
    mpg: 'video/mpeg',
    m1v: 'video/mpeg',
    m2v: 'video/mpeg',
    wmv: 'video/x-ms-wmv',
    wma: 'audio/x-ms-wma',
    avi: 'video/x-msvideo',
    m4a: 'audio/x-m4a',
    csv: 'text/csv',
    xla: 'application/vnd.ms-excel',
    xlc: 'application/vnd.ms-excel',
    xlm: 'application/vnd.ms-excel',
    xls: 'application/vnd.ms-excel',
    xlt: 'application/vnd.ms-excel',
    xlw: 'application/vnd.ms-excel',
    xlsx: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
    ods: 'application/vnd.oasis.opendocument.spreadsheet',
    gsheet: 'application/vnd.google-apps.spreadsheet',
};
