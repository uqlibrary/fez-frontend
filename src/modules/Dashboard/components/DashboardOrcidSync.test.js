import { render, waitElementToBeInDocument, WithReduxStore, WithRouter } from 'test-utils';
import DashboardOrcidSyncContainer, {
    openUrl,
    DashboardOrcidSync,
    getOnSyncPreferenceChangeHandler,
} from './DashboardOrcidSync';
import { pathConfig } from '../../../config';
import React, { useState } from 'react';
import * as actions from '../../../actions/actionTypes';
import { debounce } from 'throttle-debounce';
import { updateCurrentAuthor } from '../../../actions';

const setIsSyncEnabledMock = jest.fn();
jest.mock('react', () => {
    return {
        ...jest.requireActual('react'),
        useState: jest.fn(),
    };
});

jest.mock('throttle-debounce', () => {
    return {
        ...jest.requireActual('throttle-debounce'),
        debounce: jest.fn(),
    };
});

jest.mock('actions', () => {
    return {
        ...jest.requireActual('actions'),
        updateCurrentAuthor: jest.fn(),
    };
});

const defaultAuthor = {
    aut_id: 123,
    aut_orcid_id: '0000-1111-1111-1111',
    aut_orcid_works_last_sync: '2020-02-14 16:23:30',
    aut_is_orcid_sync_enabled: 1,
};

function setup(
    testProps = {},
    route = pathConfig.dashboard,
    routeInitialEntries = [pathConfig.dashboard],
    rerender = render,
) {
    const props = {
        ...testProps,
        author: {
            ...defaultAuthor,
            ...(testProps?.author || {}),
        },
    };

    return rerender(
        <WithReduxStore>
            <WithRouter route={route} initialEntries={routeInitialEntries}>
                <DashboardOrcidSync {...props} />
            </WithRouter>
        </WithReduxStore>,
    );
}

describe('DashboardOrcidSync', () => {
    beforeEach(() => {
        useState.mockImplementation(initialState => [initialState, setIsSyncEnabledMock]);
    });
    afterEach(() => jest.resetAllMocks());

    describe('Container', () => {
        it('should render properly', () => {
            const { container } = render(
                <WithReduxStore>
                    <WithRouter>
                        <DashboardOrcidSyncContainer author={{ aut_orcid_works_last_sync: '2020-02-14 16:23:30' }} />
                    </WithRouter>
                </WithReduxStore>,
            );
            expect(container).toMatchSnapshot();
        });
    });

    describe('Component', () => {
        it('should render properly when sync is disabled', () => {
            const { container } = setup({
                author: { aut_orcid_works_last_sync: null, aut_is_orcid_sync_enabled: 0 },
            });
            expect(container).toMatchSnapshot();
        });

        it('should render properly when sync is enabled', () => {
            const { container } = setup({});
            expect(container).toMatchSnapshot();
        });

        it('should render properly', () => {
            const { container } = setup({});
            expect(container).toMatchSnapshot();
        });

        it('should render Pending status', () => {
            const { container } = setup({
                orcidSyncStatus: {
                    orj_status: 'Pending',
                },
            });
            expect(container).toMatchSnapshot();
        });

        it('should render Done status', () => {
            const { container } = setup({
                orcidSyncStatus: {
                    orj_status: 'Done',
                },
            });
            expect(container).toMatchSnapshot();
        });

        it('should render Error status', () => {
            const { container } = setup({
                orcidSyncStatus: {
                    orj_status: 'Error',
                },
            });
            expect(container).toMatchSnapshot();
        });

        it('should update local state variable isSyncEnabled when author sync preferences change', async () => {
            const { rerender } = setup({ author: { aut_is_orcid_sync_enabled: 0 } });
            expect(setIsSyncEnabledMock).toHaveBeenCalledWith(false);
            setup({ author: { aut_is_orcid_sync_enabled: 1 } }, pathConfig.dashboard, [pathConfig.dashboard], rerender);
            expect(setIsSyncEnabledMock).toHaveBeenCalledWith(true);
        });

        it('should display confirmation dialog when route state variable `showOrcidLinkingConfirmation` is set to true', async () => {
            // `showOrcidLinkingConfirmation` is set to true upon the Orcid linking - see Orcid component
            setup({}, '/', [{ path: pathConfig.dashboard, state: { showOrcidLinkingConfirmation: true } }]);
            await waitElementToBeInDocument('orcid-sync-confirmation-message');
        });

        it('should display display progress icon next to orcid badge upon saving orcid sync preferences', async () => {
            setup({ accountAuthorSaving: true });
            await waitElementToBeInDocument('dashboard-orcid-sync-progress-icon');
        });

        it('should display display error icon next to orcid badge upon failing to save orcid sync preferences', async () => {
            setup({ accountAuthorError: true });
            await waitElementToBeInDocument('dashboard-orcid-sync-error-icon');
        });

        describe('openUrl', () => {
            it('should return function that calls window.open', () => {
                const testOpen = jest.spyOn(window, 'open');
                testOpen.mockImplementation(() => {});
                const url = 'https://www.uq.edu.au/';
                openUrl(url)();
                expect(testOpen).toHaveBeenCalledWith(url, '_blank');
            });
        });

        describe('getOnSyncPreferenceChangeHandler', () => {
            it('should bail when sync preferences already reflects the given value', () => {
                const setIsSyncEnabledMock = jest.fn();

                getOnSyncPreferenceChangeHandler(defaultAuthor, false)(true);

                expect(setIsSyncEnabledMock).not.toHaveBeenCalled();
            });

            it('should bail when sync preferences is being saved', () => {
                const setIsSyncEnabledMock = jest.fn();

                getOnSyncPreferenceChangeHandler(defaultAuthor, true)(true);

                expect(setIsSyncEnabledMock).not.toHaveBeenCalled();
            });

            it('should update local state and dispatch actions when sync preferences are being updated', () => {
                debounce.mockImplementation((delay, callback) => callback);
                const setIsSyncEnabledMock = jest.fn();
                const dispatchMock = jest.fn();
                const hideDrawerMock = jest.fn();

                getOnSyncPreferenceChangeHandler(
                    defaultAuthor,
                    false,
                    setIsSyncEnabledMock,
                    dispatchMock,
                    hideDrawerMock,
                )(false);

                expect(setIsSyncEnabledMock).toHaveBeenCalledWith(false);
                expect(dispatchMock).toHaveBeenCalledWith({ type: actions.CURRENT_AUTHOR_SAVING });
                expect(hideDrawerMock).toHaveBeenCalled();
                expect(debounce).toHaveBeenCalledWith(3000, expect.any(Function));
                expect(updateCurrentAuthor).toHaveBeenCalledWith(defaultAuthor.aut_id, {
                    ...defaultAuthor,
                    aut_is_orcid_sync_enabled: 0,
                });
            });
        });
    });
});
