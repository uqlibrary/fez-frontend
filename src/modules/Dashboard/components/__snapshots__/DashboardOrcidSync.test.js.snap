// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardOrcidSync should render Error status 1`] = `
<div>
  <button
    aria-label="There has been an error while processing a manual upload of your eSpace works to ORCID. Please try again in a few minutes or contact espace@library.uq.edu.au if you continue to experience difficulties."
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-hvz71z-MuiButtonBase-root-MuiIconButton-root"
    data-analyticsid="help-icon-orcid"
    data-mui-internal-clone-element="true"
    data-testid="help-icon-orcid"
    id="help-icon-orcid"
    tabindex="0"
    type="button"
  >
    <svg
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-pmbis9-MuiSvgIcon-root"
      data-testid="SyncProblemIcon"
      focusable="false"
      role="img"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 12c0 2.21.91 4.2 2.36 5.64L3 20h6v-6l-2.24 2.24C5.68 15.15 5 13.66 5 12c0-2.61 1.67-4.83 4-5.65V4.26C5.55 5.15 3 8.27 3 12m8 5h2v-2h-2zM21 4h-6v6l2.24-2.24C18.32 8.85 19 10.34 19 12c0 2.61-1.67 4.83-4 5.65v2.09c3.45-.89 6-4.01 6-7.74 0-2.21-.91-4.2-2.36-5.64zm-10 9h2V7h-2z"
      />
      <title>
        There has been an error while processing a manual upload of your eSpace works to ORCID. Please try again in a few minutes or contact espace@library.uq.edu.au if you continue to experience difficulties.
      </title>
    </svg>
  </button>
</div>
`;

exports[`DashboardOrcidSync should render Pending status 1`] = `
<div>
  <button
    aria-label="A manual upload of your eSpace works to ORCID is in progress."
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-hvz71z-MuiButtonBase-root-MuiIconButton-root"
    data-analyticsid="help-icon-orcid"
    data-mui-internal-clone-element="true"
    data-testid="help-icon-orcid"
    id="help-icon-orcid"
    style="margin-left: 2px; margin-bottom: 2px;"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1w1pxvu-MuiCircularProgress-root"
      font-size="small"
      role="progressbar"
      style="width: 40px; height: 40px;"
      titleaccess="A manual upload of your eSpace works to ORCID is in progress."
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`DashboardOrcidSync should render properly 1`] = `
<div>
  <button
    aria-label="Information about uploading your eSpace works to ORCID"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-hvz71z-MuiButtonBase-root-MuiIconButton-root"
    data-analyticsid="help-icon-orcid"
    data-mui-internal-clone-element="true"
    data-testid="help-icon-orcid"
    id="help-icon-orcid"
    tabindex="0"
    type="button"
  >
    <svg
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-pmbis9-MuiSvgIcon-root"
      data-testid="HelpOutlineIcon"
      focusable="false"
      role="img"
      viewBox="0 0 24 24"
    >
      <path
        d="M11 18h2v-2h-2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8m0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4"
      />
      <title>
        Information about uploading your eSpace works to ORCID
      </title>
    </svg>
  </button>
</div>
`;

exports[`DashboardOrcidSyncContainer should render properly 1`] = `
<div>
  <button
    aria-label="Information about uploading your eSpace works to ORCID"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-hvz71z-MuiButtonBase-root-MuiIconButton-root"
    data-analyticsid="help-icon-orcid"
    data-mui-internal-clone-element="true"
    data-testid="help-icon-orcid"
    id="help-icon-orcid"
    tabindex="0"
    type="button"
  >
    <svg
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-pmbis9-MuiSvgIcon-root"
      data-testid="HelpOutlineIcon"
      focusable="false"
      role="img"
      viewBox="0 0 24 24"
    >
      <path
        d="M11 18h2v-2h-2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8m0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4"
      />
      <title>
        Information about uploading your eSpace works to ORCID
      </title>
    </svg>
  </button>
</div>
`;
